src/devices/grill_thermometer.c

```diff
@@ -0,0 +1,109 @@
+/** @file
+    Remote Grill Thermometer temperature sensor.
+
+    Copyright (C) 2023 Ethan Halsall
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+/** @fn int grill_thermometer_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+Remote Grill Thermometer -- Generic wireless thermometer with probe.
+
+This is a meat thermometer with no brand / model identification except the FCC ID.
+
+Manufacturer:
+- Yangzhou Fupond Electronic Technology Corp., Ltd
+
+Supported Models:
+- RF-T0912 (FCC ID TXRFPT0912)
+
+9 - 415 F, frequency 434.052 MHz
+
+Data structure:
+
+10 repetitions of the same 24 bit payload.
+
+    AAAAAAAA AAAAAAAA BBBBBBBB
+
+- A: 16 bit temperature in Fahrenheit. Big Endian.
+- B: Checksum of A
+
+*/
+
+#include "decoder.h"
+
+static int grill_thermometer_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    short temp_f     = 0;
+    int overload     = 0;
+    int repeats      = 0;
+    uint8_t checksum = 0;
+
+    bitbuffer_invert(bitbuffer);
+
+    // use the most recent "valid" data that repeats more than once
+    for (int row = 0; row < bitbuffer->num_rows; row++) {
+        uint8_t *row_data = bitbuffer->bb[row];
+        checksum          = row_data[0] + row_data[1];
+
+        if (bitbuffer->bits_per_row[row] != 24 ||
+                checksum != row_data[2] ||
+                checksum == 0) {
+            continue;
+        }
+
+        short current_value = (row_data[0] << 8) + row_data[1];
```

Comment:
we usually write this as:
`int current_value = (int16_t)(row_data[0] << 8) | row_data[1]; // uses sign-extend`